drivers/hmis/spec/models/hmis/ce/match_engine_spec.rb

```diff
@@ -28,60 +19,161 @@
     )
   end
 
-  let(:clients) do
-    Hmis::Hud::Client.where(id: [
-      client_adult_non_veteran,
-      client_minor_non_veteran,
-      client_adult_veteran,
-      client_senior_veteran,
-    ].map(&:id))
-  end
-
-  let(:adult_clients) do
-    clients - [client_minor_non_veteran]
-  end
+  # Override in specific tests
+  let(:requirement_expression) { 'TRUE' }
+  let(:priority_expression) { '0' }
 
   def generate_candidates(pool, clients)
+    # create destination clients
+    GrdaWarehouse::Tasks::IdentifyDuplicates.new.run!
     described_class.call(pool, clients)
     pool.candidates
   end
 
-  describe 'Policy that is always false' do
-    let(:requirement_expression) { '1=0' }
-    it 'returns no candidates' do
-      results = generate_candidates(pool, clients)
-      expect(results).to be_empty
+  shared_context 'with demographic test clients' do
+    let(:client_adult_non_veteran) { create(:hmis_hud_client, veteran_status: 0, dob: 20.years.ago) }
+    let(:client_minor_non_veteran) { create(:hmis_hud_client, veteran_status: 0, dob: 10.years.ago) }
+    let(:client_adult_veteran) { create(:hmis_hud_client, veteran_status: 1, dob: 20.years.ago) }
+    let(:client_senior_veteran) { create(:hmis_hud_client, veteran_status: 1, dob: 68.years.ago) }
+
+    let(:all_clients) do
+      [client_adult_non_veteran, client_minor_non_veteran, client_adult_veteran, client_senior_veteran]
```

Comment:
Not sure exactly where to put it, but I think it would be nice to test these expressions against a client with missing dob (and/or missing veteran status), so we have that expectation written out somewhere. (Eg `current_age > 18` excludes clients with missing dob, I think)